runtime/doc/lua.txt

```diff
@@ -1778,4 +1778,77 @@ select({items}, {opts}, {on_choice})                         *vim.ui.select()*
                                  1-based index of `item` within `item` . `nil`
                                  if the user aborted the dialog.
 
+
+==============================================================================
+Lua module: keymap                                                *lua-keymap*
+
+del({modes}, {lhs}, {opts})                                 *vim.keymap.del()*
+                Remove an existing keymap. Examples: >
+
+                   vim.keymap.del('n', 'lhs')
+
+                   vim.keymap.del({'n', 'i', 'v'}, '<leader>w', { buffer = 5 })
+<
+
+                Parameters: ~
+                    {opts}  table A table of optional arguments:
+                            • buffer: Can be boolean or bufnr . Needed when
+                              removing buffer local keymaps. buffer true or 0
+                              refers to current buffer.
+
+                See also: ~
+                    |vim.keymap.set()|
+
+set({mode}, {lhs}, {rhs}, {opts})                           *vim.keymap.set()*
+                Add a new keymap. Examples: >
+
+                   -- You can make keymaps worth Lua functions
+                   vim.keymap.set('n', 'lhs', function() print("real lua function") end)
+
+                   -- Can use it to map multiple modes
+                   vim.keymap.set({'n', 'v'}, '<leader>lr', vim.lsp.buf.references, { silent = true, buffer=true })
+
+                   -- You can map a key for specific buffer
+                   vim.keymap.set('n', '<leader>w', "<cmd>w<cr>", { silent = true, buffer = 5 })
+
+                   -- Also expr keymaps too. Notice the replace_termcodes when making Lua expr maps.
+                   vim.keymap.set('i', '<Tab>', function()
+                     return vim.fn.pumvisible() == 1 and "<C-n>" or "<Tab>"
+                   end, { silent = true, expr = true, replace_termcodes = true })
+                   -- While setting <Plug> mapping remember to set remap to true
+                   vim.keymap.set('n', '[', '<Plug>(MatchitNormalMultiBackward)', {silent = true, remap = true })
+<
+
+                A pit fall of this is if you want to make keymap like: >
+
+                    vim.keymap.set('n', 'asdf', require('jkl').my_fun)
+<
+
+                `require('jkl)` get's evaluated durimg this call. If you want to avoid this
+                cost at startup you can wrap it in a function, for example: >
+
+                    vim.keymap.set('n', 'asdf', function() return require('jkl').my_fun() end)
+<
+
+                Parameters: ~
+                    {mode}  string|table Same mode short names as
+                            |nvim_set_keymap()|. Can also be list of modes to
+                            create keymap on multiple modes.
+                    {lhs}   string Left-hand-side |{lhs}| of the mapping.
+                    {rhs}   string|function Right-hand-side |{rhs}| of the
```

Comment:
Hmm looks like we can't use `|` as the type separator here since it's interpreted as a tag by the help syntax. I'm not sure what to use instead, any ideas?

Reply:
How about '/' ? Though I think emmy ;ua notation uses '|' .

Reply:
Huh . It's actually alright . Open the file in neovim it gets displayed correctly . Githubs highlighting is broken.